Content.Server/Objectives/Objective.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using Content.Server.Mobs;
+using Content.Server.Objectives.Interfaces;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.Objectives
+{
+    public class Objective : IEquatable<Objective>
+    {
+        [ViewVariables]
+        public readonly Mind Mind;
+        [ViewVariables]
+        public readonly ObjectivePrototype Prototype;
+        private List<IObjectiveCondition> _conditions = new List<IObjectiveCondition>();
+        [ViewVariables]
+        public IReadOnlyList<IObjectiveCondition> Conditions => _conditions;
+
+        public Objective(ObjectivePrototype prototype, Mind mind)
+        {
+            Prototype = prototype;
+            Mind = mind;
+            foreach (var condition in prototype.Conditions)
+            {
+                _conditions.Add(condition.GetAssigned(mind));
+            }
+        }
+
+        public bool Equals(Objective other)
+        {
+            if (ReferenceEquals(null, other)) return false;
+            if (ReferenceEquals(this, other)) return true;
+            if (!Equals(Mind, other.Mind) || !Equals(Prototype, other.Prototype)) return false;
+            if (_conditions.Count != other._conditions.Count) return false;
+            for (int i = 0; i < _conditions.Count; i++)
```

Comment:
```suggestion
            for (var i = 0; i < _conditions.Count; i++)
```
But you could also just make this a foreach loop?

Reply:
because i am also accessing `other._conditions`